public/calendar_reboot.js

```diff
@@ -921,151 +933,187 @@ var LocalCalendarStorage = function(params) {
       
     };
     
-    var jitter = function(func, arg) {
-        var runFunc = function () {
-            func(arg);
-            self.endWork();
-        };
-        
-        var randomNumber = Math.random() * 4000;
-        self.startWork();
-        setTimeout(runFunc, randomNumber);
-        
-    };
-    
     self.getAllCalendarIds = function() {
         // Returns a promise for the allCalendarIds object from storage
         
-        return new Promise( function(resolve, reject) {
-            var allCalendarIds = loadFromLocalStorage(toKey(allCalendarIdsKey));
-            
-            if (allCalendarIds !== null ) {
-
-                jitter(resolve, allCalendarIds);
-
-            }
-            else {
-                jitter(reject, "Not found");
-            }
-        })
-        .then( function(ids) {
-            return ids;
+        var userId = self.user.uid;
+        self.startWork();
+        return new Promise(function(resolve, reject) {
+            self.database.ref('/users/' + userId + '/allCalendarIds')
+            .once('value')  
+            .then(function(allCalendarIds) {
+                self.endWork();
+                if (allCalendarIds.val() === null) {
+                    reject("allCalendarIds not found");
+                }
+                else {
+                    return resolve(allCalendarIds.val());
+                }
+            })
+            .catch(function(err) {
+                self.endWork();
+                console.error(err);
+                return err;
+            })
         })
-        .catch( function() {
-            console.log("getAllCalendarIds catch function running.");
-        });
-
     };
     
     self.save = function(calendarObj) {
         //save an App object (like a calendar object for example) in storage
         
-        //store the state in localStorage
-
-        var stateP = new Promise(function(resolve, reject) {
-            storeInLocalStorage(toKey(calendarObj.state.uniqueId), calendarObj.state);
-            jitter(resolve);
-        });
+        // Store the calendar state in the firebase database.
+        // TO DO save the actual calendar state.
         
+        var userId = self.user.uid;
+        var calUniqueId = calendarObj.state.uniqueId;
+        var calTitle = calendarObj.state.title;
+        var calState = calendarObj.state;
         
-        //put calendar in allCalendarIdss and store it
+        // Store the user's allCalendarIds and their calendarState .. setting the
+        // active calendar has it's own method.
```

Comment:
You don't have to save the currentActiveCalendar in the save method because setActive is its own method. I was just writing that there so that I could remember that.